doc/plugin_tutorial/tuto4/src/myexternals.ml

```diff
@@ -0,0 +1,122 @@
+open Names
+open Ltac2_plugin
+open Tac2externals
+open Tac2ffi
+
+(* convenience alias *)
+let return = Proofview.tclUNIT
+
+(* used to distinguish our primitives from some other plugin's primitives
+   by convention matches the plugin's ocamlfind name *)
+let plugin_name = "rocq-plugin-tutorial.tuto4"
+
+let pname s = { Tac2expr.mltac_plugin = plugin_name; mltac_tactic = s }
+
+(* convenience wrapper around Tac2externals.define *)
+let define s = define (pname s)
+
+(* a simple function taking an integer and returning a boolean
+   "@->" is an infix function from Tac2externals combining "type representation" (Tac2ffi.repr)
+   "ret" means we return the answer without doing tactic operations (no access to the Proofview monad) *)
+let () = define "the_question" (int @-> ret bool) @@ fun i ->
+  Int.equal i 42
+
+(* a wrapper around "exact", it takes a constr (ie a term) and returns nothing
+   "tac" means we have access to the tactic monad *)
+let () = define "my_exact" (constr @-> tac unit) @@ fun c ->
+  Tactics.exact_check c
+
+(** **** Transparent custom type *)
+
+(* We define a custom type in ocaml and in ltac2 (this is the ocaml side) *)
+type my_custom =
+  | A
+  | B of EConstr.t
+
+(* translate from ocaml to the ltac2 representation (Tac2val.valexpr)
+   we use Tac2ffi.of_constr in the B case *)
+let of_custom = function
+  | A -> of_int 0
+  | B c -> of_block (0, [|of_constr c|])
+
+(* Go from the ltac2 representation to the ocaml representation.
+   This needs to look at the low-level valexpr data.
+   If an external is declared with an incorrect Ltac2 type it may be passed invalid values,
+   in which case we assert false. *)
```

Comment:
```suggestion
(* Go from the Ltac2 representation to the OCaml representation.
   This needs to look at the low-level valexpr data.
   If an external is declared with an incorrect Ltac2 type it may be passed
   invalid values, in which case we assert false.
   ValInt (for an integer) and ValBlk (for a block) are two constructors for
   Ltac2 representations (valexprs). *)
```